### ISSUE TITLE
feat: Add `shorebird build` command to build an app with the shorebird engine **without** making a new release

### ISSUE THREAD

**Issue by mtwichel** [shorebirdtech/shorebird#1030]
**Description**

As a developer of white label apps, we release many applications that share 100% of their Dart code (as customization of apps happens at run time, not build time). As a result, we would like to have one application in shorebird that all our apps share so that when a patch is required, we only need to run `shorebird patch` once. See this discord thread for more info: https://discord.com/channels/1030243211995791380/1135674823872364685

What is currently blocking this behavior is that the only way to build an app that includes the shorebird engine is `shorebird release`. However, that command not only builds the app, but pushes a new release to shorebird's servers. That leads us into this scenario:

1. We build app A with `shorebird release`, deploy it to the App Store
2. No code is added to the repository
3. We build app B with `shorebird release`, deploy it to the app store
4. Now app A and app B are on different releases, even if they share exactly the same Dart code

Basically, I would like to be able to run a command (maybe `shorebird build`) that just builds the application (either iOS or Android) with the shorebird engine and doesn't submit a release. I would expect it to work just like `shorebird preview` in the sense that it takes the more recent patch for the given release.

_Side Note:_ I fully expect to have to run `shorebird release` when needed, but ideally we only have to do that once per release, not for every client.

LMK what you think of this idea, and if I submitted improperly I'm truly sorry; I looked for a CONTRIBUITING.md file to see if opening these kinds of issues was okay, but didn't see one so I went for it :)

Thanks!

**Comment by felangel** [comment]
Hi @mtwichel 👋 
Thanks for opening an issue!

We currently support that via:

- `shorebird build apk`
- `shorebird build aar`
- `shorebird build appbundle`
- `shorebird build ipa`

Let us know if that helps 👍

**Comment by mtwichel** [comment]
Well look at that! I'll try it!

As a side note - I'm not sure if that exists in the docs anywhere. It's possible I'm just blind though 😂

**Comment by eseidel** [comment]
Let us know if this doesn't help!
